### ISSUE TITLE
Vampiric Embrace has weird interactions with the threat table

### ISSUE THREAD

**Issue by suggney** [slowtorta/turtlewow-bug-tracker#2334]
**Please, provide a clear description what the bug is:**

Vampiric Embrace seems to not generate any threat in situations where it should, albeit unreliably. Sometimes it works.

**Steps to reproduce the behavior with as much detail as possible:**

1. Have TWThreat enabled
2. Cast Vampiric Embrace on a target
3. Experiment! The threat generated seems to be totally unreliable, and changes based on things like how many people are healing from Vampiric Embrace.

**IDs of affected creatures, items, quests or spells with a link to the relevant page:**

Spell: https://database.turtle-wow.org/?spell=15286

**Expected behavior. Describe how it should work:**

Vampiric Embrace heals 20% (30% if talented) of all damage you dealt back to you and your party members. For each hit point healed, 0.5 threat should be generated on you. For example, if I deal 100 damage, I will heal 30 to myself and generate 15 threat, in addition to a cumulative 120 to my four party members and generate 60 threat.

**Comment by balakethelock** [comment]
Healing threat is split on all mobs currently in combat with you. That means if you are fighting two mobs at the same time, healing  for 150 will generate 75 split between those two mobs, 37.5 on each one.

**Comment by suggney** [comment]
> Healing threat is split on all mobs currently in combat with you. That means if you are fighting two mobs at the same time, healing for 150 will generate 75 split between those two mobs, 37.5 on each one.

Yes, I know. Vampiric Embrace seems to be generating no threat in some double pulling situations.

**Comment by Platinumcoated** [comment]
Also keep in mind that only the effective healing done generates threat, overhealing doesn't
